### PR TITLE
Implement slicing for arrays

### PR DESCRIPTION
**Pull request by hoodmane**
More work on features discussed in #2903.

### CODE REVIEW

File: src/core/jsproxy.c [hunk 1]
```diff
@@ -467,9 +467,27 @@ JsProxy_subscript_array(PyObject* o, PyObject* item)
     return pyresult;
   }
   if (PySlice_Check(item)) {
-    PyErr_SetString(PyExc_NotImplementedError,
-                    "Slice subscripting isn't implemented");
-    return NULL;
+    Py_ssize_t start, stop, step, slicelength;
+    if (PySlice_Unpack(item, &start, &stop, &step) < 0) {
```
Comment (ryanking13) [inline src/core/jsproxy.c:471]: I think we can use `FAIL_IF_MINUS_ONE` here too, same as what you did in `JsArray_ass_subscript`.

**Review by ryanking13 — APPROVED** [review]
Thanks, @hoodmane. Looks very cool. Please update the changelog, otherwise LGTM.

File: src/core/jsproxy.c (hunk 1 above)
Comment (hoodmane) [reply src/core/jsproxy.c:471]: Can't because there is no `finally` block in this function. Perhaps we should add one that just returns `NULL`.

Comment (hoodmane) [reply src/core/jsproxy.c:471]: I adulterated my typical C style here with some code that was nearly copied verbatim from `cpython/Objects/listobject.c`

**Comment by rth** [comment]
Very nice!

**Pull request merged**